src/theory/quantifiers/term_enumeration.cpp

```diff
@@ -109,15 +109,18 @@ bool TermEnumeration::mayComplete(TypeNode tn)
   if (it == d_may_complete.end())
   {
     bool mc = false;
-    if (isClosedEnumerableType(tn) && tn.getCardinality().isFinite()
-        && !tn.getCardinality().isLargeFinite())
+    if (isClosedEnumerableType(tn) && tn.isInterpretedFinite())
     {
-      Node card = NodeManager::currentNM()->mkConst(
-          Rational(tn.getCardinality().getFiniteCardinality()));
-      Node oth = NodeManager::currentNM()->mkConst(Rational(1000));
-      Node eq = NodeManager::currentNM()->mkNode(LEQ, card, oth);
-      eq = Rewriter::rewrite(eq);
-      mc = eq.isConst() && eq.getConst<bool>();
+      Cardinality c = tn.getCardinality();
+      if( !c.isLargeFinite() )
+      {
+        Node card = NodeManager::currentNM()->mkConst(
+            Rational(c.getFiniteCardinality()));
+        Node oth = NodeManager::currentNM()->mkConst(Rational(1000));
```

Comment:
what about doing an option instead of this hardcoded limit? It'd be cool to be able to use FMF on bitvectors of bitwidth larger than 9.